lib/IDE/CodeCompletion.cpp

```diff
@@ -1240,7 +1240,7 @@ CodeCompletionResult *CodeCompletionResultBuilder::takeResult() {
       typeRelation =
           calculateMaxTypeRelationForDecl(AssociatedDecl, declTypeContext);
 
-    if (typeRelation == CodeCompletionResult::Invalid) {
+    if (!IsNotRecommended && typeRelation == CodeCompletionResult::Invalid) {
       IsNotRecommended = true;
       NotRecReason = CodeCompletionResult::NotRecommendedReason::TypeMismatch;
     }
```

Comment:
At this point, I don't remove this because `complete.open` uses this for sorting, and I don't want to change them to use the type relation. It'd be non trivial change.